# Test Yourself

1. Which of the following is not true regarding sequence mapping?
* Mainly useful for finding exact matches
* Can be used for mapping short NGS reads
* Requires a method for scoring sequence alignments
* Performs poorly on repetitive sequences

2. Which of the following is true regarding the effect of Blast parameters? (mark all that apply)
* Increasing W (word size) will decrease the number of HSPs
* Increasing X (drop-off) will increase the length of HSPs
* Increasing W (word size) will increase the number of HSPs
* Increasing X (drop-off) will decrease the length of HSPs
* Increasing X (drop-off) will increase the number of HSPs
* Increasing X (drop-off) will decrease the number of HSPs

3. In which of the following situations would you use local alignment?
* Mapping short NGS reads from an african individual to the human reference genome to detect SNPs
* Mapping an mRNA sequences of two splice variants to a genome to detect differences in splice sites
* Search a DB for proteins of the same family, based on the presence of protein domains
* Finding sequence differences between homologous genes of mouse and rat
* Infer the phylogenetic tree of 10 bacteria strains based on rRNA sequence

4. Which BLAST program would you use to find protein coding regions within a whole genome sequence, based on protein sequences of a closely-related species?
* ```TBLASTN```
* ```BLASTN```
* ```BLASTP```
* ```BLASTX```
* ```TBLASTX```
